### PR TITLE
Assert more pattern constraints on class-linking slots

### PR DESCRIPTION
**Pull request by kheal**
This PR adds pattern constraints to several classes through asserting `slot_usage` and updates associated tests to pass these tests (previously using non-passing IDs).

Addresses https://github.com/microbiomedata/nmdc-schema/issues/2061.

1) `ProtocolExecution` now has pattern constraints for `has_input`, `has_output`, and `has_process_parts` slots
2) `StorageProcess` now has pattern constraints for `id` slot.  (I had missed this class during https://github.com/microbiomedata/berkeley-schema-fy24/pull/177)
3) `MassSpectrometry` now has pattern constraints for `has_calibration`, `has_chromatography_configuration`, and `has_mass_spectrometry_configuration` slots.  

No migration needed and no anticipated effects on refactoring process, this is just shoring up specificity in the model to prevent future id issues.

### CODE REVIEW

**Review by turbomam — APPROVED** [review]

**Comment by turbomam** [comment]
@aclum do you need more time to review? I think we should merge this in ASAP.

**Comment by aclum** [comment]
@turbomam I am looking at this now. Will be done shortly.

File: src/schema/nmdc.yaml [hunk 1]
```diff
@@ -408,6 +420,20 @@ classes:
         structured_pattern:
           syntax: "{id_nmdc_prefix}:pex-{id_shoulder}-{id_blade}$"
           interpolated: true
+      has_input:
+        structured_pattern:
+          syntax: "{id_nmdc_prefix}:(bsm|procsm)-{id_shoulder}-{id_blade}$"
+          interpolated: true
+      has_output:
+        structured_pattern:
+          syntax: "{id_nmdc_prefix}:(bsm|procsm)-{id_shoulder}-{id_blade}$"
```
Comment (aclum) [inline src/schema/nmdc.yaml:429]: @kheal we are allowing has_output to be a biosample? @mslarae13

**Comment by aclum** [comment]
@kheal @turbomam   this is still missing some classes where slots are inherited. See DissolvingProcess, it inherits `has_input` and `has_output` from PlannedProcess so I think was missed by the checks you were using.

**Comment by kheal** [comment]
> @kheal @turbomam this is still missing some classes where slots are inherited. See DissolvingProcess, it inherits `has_input` and `has_output` from PlannedProcess so I think was missed by the checks you were using.

My understanding is that bc `DissolvingProcess` is_a `MaterialProcessing`, it inherits the slot usage from `MaterialProcessing`, which is applicable.  I'll double check this in a test, but that's what I would expect.

**Comment by aclum** [comment]
@kheal you are right, disregard that comment. I do still think that has_output for ProtocolExecution should only be a biosample b/c MaterialProcessing has_output is always a ProcessedSample

**Comment by kheal** [comment]
@aclum - Fixed the ProtocolExecution `has_output` constraint, thanks for the eagle eyes.

**Review by aclum — APPROVED** [review]

**Pull request merged**